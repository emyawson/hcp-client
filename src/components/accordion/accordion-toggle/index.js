export * from './accordion-toggle.component';
