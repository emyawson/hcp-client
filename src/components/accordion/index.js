export * from './accordion.component';
export * from './accordion-item';
export * from './accordion-toggle';
