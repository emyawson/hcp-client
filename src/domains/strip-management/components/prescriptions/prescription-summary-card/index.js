export * from './prescription-summary-card.component';
export * from './prescription-summary-card-with-loader.component';
