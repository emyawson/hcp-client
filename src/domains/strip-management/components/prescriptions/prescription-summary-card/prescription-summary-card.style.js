import styled from 'styled-components';

import { breakpoints, colors, fontSize, spacing } from 'src/core';
import { weight } from 'src/components/fonts/weights';
import { convertPxToRem, combineRems } from 'src/utils';

const colorBorder = colors.silver;

export const PrescriptionInfoTitle = styled.span`
  color: ${colors.grayMedium};
  font-size: ${fontSize.caption};
  font-weight: ${weight.bold};
  text-transform: uppercase;
  display: flex;
  flex-direction: column;
`;

export const PrescriptionSubHeading = styled.span`
  color: ${colors.grayDark};
  font-size: ${fontSize.caption};
  font-weight: ${weight.bold};
  margin-top: ${spacing.two};
  text-transform: uppercase;
  display: flex;
  flex-direction: column;
`;

export const PrescriptionSummaryCardContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  line-height: 1;
  width: 100%;
`;

export const PrescriptionSummaryCardHeaderDiv = styled.div`
  align-items: center;
  background: ${props => (props.active ? colors.lavender : colors.silverLight)};
  border-right: 1px solid
    ${props => (props.active ? colors.quartzBlue : colorBorder)};
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-evenly;
  margin-bottom: ${props => (props.isNested && props.active ? '-1px' : '0')};
  min-width: ${convertPxToRem(150)};
  min-height: ${spacing.five};
`;

export const PrescriptionSummaryCardHeader = styled.h1`
  color: ${colors.grayDark};
  display: flex;
  flex-direction: column;
  font-size: ${fontSize.caption};
  font-weight: ${weight.semiBold};
  text-transform: uppercase;
`;

export const PrescriptionSummaryCardSubHeading = styled.h2`
  color: ${colors.brandBlue};
  display: flex;
  flex-direction: column;
  font-size: ${fontSize.caption};
  font-weight: ${weight.bold};
  margin-top: ${spacing.two};
  text-transform: uppercase;
`;

export const PrescriptionSummaryCardLineItem = styled.div`
  border: none;
  color: ${colors.grayDark};
  font-size: ${fontSize.p};
  font-weight: ${weight.light};
  margin: 0;
  padding: ${spacing.one} ${combineRems(spacing.two, spacing.three)};
  text-align: left;
  box-shadow: 12px 0px 1px -12px ${colors.grayMedium};

  &:first-child {
    box-shadow: none;
  }

  @media (min-width: ${breakpoints.large}) {
    padding: ${spacing.two} ${spacing.four};
  }
`;
export const PrescriptionSummaryCardLineItemNoInfo = PrescriptionSummaryCardLineItem.extend`
  text-align: center;
  width: 100%;
`;

export const PrescriptionSummaryCardLineItemBlock = styled.span`
  display: block;
`;
