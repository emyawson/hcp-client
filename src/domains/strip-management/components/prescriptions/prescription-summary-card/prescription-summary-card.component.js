import React from 'react';
import { equals, not } from 'ramda';

import { hasValue, RenderIf } from 'src/utils';
import { translate } from 'src/i18n';
import {
  isTypePermanent,
  PATIENT_PRESCRIPTION_TYPES,
} from 'src/core/prescription';
import { LocalizedText } from 'src/components';
import { AccordionTitleDiv } from 'src/components/accordion/accordion.style';

import {
  PrescriptionSummaryCardContainer,
  PrescriptionSummaryCardHeaderDiv,
  PrescriptionSummaryCardHeader,
  PrescriptionSummaryCardLineItem,
  PrescriptionSummaryCardLineItemNoInfo,
  PrescriptionSummaryCardSubHeading,
  PrescriptionSubHeading,
  PrescriptionInfoTitle,
} from './prescription-summary-card.style';

const prescriptionTitles = {
  permanent: translate('prescription.permanentTitle'),
  temporary: translate('prescription.temporaryTitle'),
};

const generateStripConsumptionDisplayString = () =>
  `${translate('prescription.stripConsumption.title')}`;

const generateStripConsumptionDisplayFormulaString = ({ period, quantity }) => {
  const periodString = translate(`prescription.period.${period}`);
  return `${quantity} x ${periodString}`;
};

const generatePatientStockString = patientStock =>
  `${patientStock} ${translate('prescription.patientStock.units')}`;

export const PrescriptionSummaryCard = ({
  frequency = '',
  nextDeliveryDate = '',
  stripModel = '',
  titleKey,
  active = false,
  contentOnly = false,
  patientStock,
  isUnsaved = false,
  prescriptionType = PATIENT_PRESCRIPTION_TYPES.PERMANENT,
  period,
  quantity,
  isAccordionHeader = false,
}) => {
  const showPrescriptionTypeTitle =
    equals(prescriptionType, PATIENT_PRESCRIPTION_TYPES.TEMPORARY) ||
    not(isUnsaved);
  const prescriptionSummaryCardContents = () => (
    <PrescriptionSummaryCardContainer
      isUnsaved={isUnsaved}
      prescriptionType={prescriptionType}
      frequency={frequency}
      nextDeliveryDate={nextDeliveryDate}
    >
      <PrescriptionSummaryCardHeaderDiv
        active={active}
        isNested={isAccordionHeader}
      >
        <PrescriptionSummaryCardLineItem>
          <PrescriptionSummaryCardHeader>
            <LocalizedText textKey="prescription.title" />
          </PrescriptionSummaryCardHeader>
          <RenderIf validate={showPrescriptionTypeTitle}>
            <PrescriptionSummaryCardSubHeading>
              {prescriptionTitles[prescriptionType] || (
                <LocalizedText textKey={titleKey} />
              )}
            </PrescriptionSummaryCardSubHeading>
          </RenderIf>
        </PrescriptionSummaryCardLineItem>
      </PrescriptionSummaryCardHeaderDiv>
      <RenderIf validate={quantity && period && not(isUnsaved)}>
        <PrescriptionSummaryCardLineItem>
          <PrescriptionInfoTitle>
            {generateStripConsumptionDisplayString()}
          </PrescriptionInfoTitle>
          <PrescriptionSubHeading>
            {generateStripConsumptionDisplayFormulaString({
              period,
              quantity,
            })}
          </PrescriptionSubHeading>
        </PrescriptionSummaryCardLineItem>
      </RenderIf>
      <RenderIf validate={nextDeliveryDate && not(isUnsaved)}>
        <PrescriptionSummaryCardLineItem>
          <PrescriptionInfoTitle>
            {translate('stripDelivery.nextDeliveryDate')}
          </PrescriptionInfoTitle>
          <PrescriptionSubHeading>{nextDeliveryDate}</PrescriptionSubHeading>
        </PrescriptionSummaryCardLineItem>
      </RenderIf>
      <RenderIf validate={stripModel && not(isUnsaved)}>
        <PrescriptionSummaryCardLineItem>
          <PrescriptionInfoTitle>
            {translate('prescription.stripModel.label')}
          </PrescriptionInfoTitle>
          <PrescriptionSubHeading>{stripModel}</PrescriptionSubHeading>
        </PrescriptionSummaryCardLineItem>
      </RenderIf>

      <RenderIf validate={hasValue(patientStock) && not(isUnsaved)}>
        <PrescriptionSummaryCardLineItem>
          <PrescriptionInfoTitle>
            {translate('prescription.patientStock.label')}
          </PrescriptionInfoTitle>
          <PrescriptionSubHeading>
            {generatePatientStockString(patientStock)}
          </PrescriptionSubHeading>
        </PrescriptionSummaryCardLineItem>
      </RenderIf>
      <RenderIf validate={isUnsaved && isTypePermanent(prescriptionType)}>
        <PrescriptionSummaryCardLineItemNoInfo>
          {translate('prescription.patientHasNoCurrentPrescription')}
        </PrescriptionSummaryCardLineItemNoInfo>
      </RenderIf>
    </PrescriptionSummaryCardContainer>
  );
  if (contentOnly) {
    return <div>{prescriptionSummaryCardContents()}</div>;
  }
  return (
    <AccordionTitleDiv active={active} noPadding={true}>
      {prescriptionSummaryCardContents()}
    </AccordionTitleDiv>
  );
};
