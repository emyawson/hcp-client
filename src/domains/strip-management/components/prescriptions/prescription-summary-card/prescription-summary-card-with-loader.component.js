import React from 'react';
import { equals } from 'ramda';

import { Banner, Link, withLoaderErrorOnTimeout } from 'src/components';
import { stripManagementInnerRoutes } from 'src/domains/strip-management/routes';
import { translate } from 'src/i18n';
import { WithPermissions } from 'src/utils/with-permissions/with-permissions.container';
import { PERMISSIONS } from 'src/core/permissions/permissions.constants';

import { PrescriptionSummaryCard } from './prescription-summary-card.component';

const emptyPrescriptionText = (
  <React.Fragment>
    {`${translate('stripDelivery.empty.prescription')} `}
    <WithPermissions hasPermissions={[PERMISSIONS.STRIP_PRESCRIPTIONS]}>
      <Link to={stripManagementInnerRoutes.prescriptionInfoByPatient}>
        {translate('stripDelivery.empty.prescriptionCta')}
      </Link>
    </WithPermissions>
  </React.Fragment>
);
const errorOptions = {
  ErrorComponent: () => <Banner text={emptyPrescriptionText} />,
};

const loaderOptions = {
  loaderProps: {
    infinite: true,
    minHeight: 48,
    size: 24,
  },
  validators: {
    active: equals(true),
  },
};

export const PrescriptionSummaryCardWithLoaderAndError = withLoaderErrorOnTimeout(
  {
    errorOptions,
    loaderOptions,
  },
)(PrescriptionSummaryCard);
