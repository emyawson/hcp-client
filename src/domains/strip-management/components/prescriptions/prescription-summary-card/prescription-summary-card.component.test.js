import React from 'react';
import { shallow } from 'enzyme';

import { AccordionTitleDiv } from 'src/components/accordion/accordion.style';

import { PrescriptionSummaryCard } from './prescription-summary-card.component';

describe('Prescription summary card component', () => {
  const mockBaseProps = {
    titleKey: 'title',
    patientStock: 100,
    period: 'days',
    quantity: 6,
  };

  it('Renders correctly', () => {
    const wrapper = shallow(<PrescriptionSummaryCard {...mockBaseProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Does not render title if card is content-only', () => {
    const mockProps = {
      ...mockBaseProps,
      contentOnly: true,
    };
    const wrapper = shallow(<PrescriptionSummaryCard {...mockProps} />);
    expect(wrapper.find(AccordionTitleDiv)).toHaveLength(0);
  });
});
