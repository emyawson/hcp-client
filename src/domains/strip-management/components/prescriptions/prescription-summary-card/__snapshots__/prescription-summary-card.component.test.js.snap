// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prescription summary card component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    patientStock={100}
    period="days"
    quantity={6}
    titleKey="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "active": false,
      "children": <styled.div
        frequency=""
        isUnsaved={false}
        nextDeliveryDate=""
        prescriptionType="permanent"
      >
        <styled.div
          active={false}
          isNested={false}
        >
          <styled.div>
            <styled.h1>
              <LocalizedText
                textKey="prescription.title"
              />
            </styled.h1>
            <f1
              validate={true}
            >
              <styled.h2>
                PRIMARY
              </styled.h2>
            </f1>
          </styled.div>
        </styled.div>
        <f1
          validate={true}
        >
          <styled.div>
            <styled.span>
              Strip Consumption
            </styled.span>
            <styled.span>
              6 x Day
            </styled.span>
          </styled.div>
        </f1>
        <f1
          validate=""
        >
          <styled.div>
            <styled.span>
              Next delivery:
            </styled.span>
            <styled.span>
              
            </styled.span>
          </styled.div>
        </f1>
        <f1
          validate=""
        >
          <styled.div>
            <styled.span>
              Model
            </styled.span>
            <styled.span>
              
            </styled.span>
          </styled.div>
        </f1>
        <f1
          validate={true}
        >
          <styled.div>
            <styled.span>
              Total Patient Stock
            </styled.span>
            <styled.span>
              100 UDS
            </styled.span>
          </styled.div>
        </f1>
        <f1
          validate={false}
        >
          <styled.div>
            No information on file. Complete new patient form below.
          </styled.div>
        </f1>
      </styled.div>,
      "noPadding": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            active={false}
            isNested={false}
          >
            <styled.div>
              <styled.h1>
                <LocalizedText
                  textKey="prescription.title"
                />
              </styled.h1>
              <f1
                validate={true}
              >
                <styled.h2>
                  PRIMARY
                </styled.h2>
              </f1>
            </styled.div>
          </styled.div>,
          <f1
            validate={true}
          >
            <styled.div>
              <styled.span>
                Strip Consumption
              </styled.span>
              <styled.span>
                6 x Day
              </styled.span>
            </styled.div>
          </f1>,
          <f1
            validate=""
          >
            <styled.div>
              <styled.span>
                Next delivery:
              </styled.span>
              <styled.span>
                
              </styled.span>
            </styled.div>
          </f1>,
          <f1
            validate=""
          >
            <styled.div>
              <styled.span>
                Model
              </styled.span>
              <styled.span>
                
              </styled.span>
            </styled.div>
          </f1>,
          <f1
            validate={true}
          >
            <styled.div>
              <styled.span>
                Total Patient Stock
              </styled.span>
              <styled.span>
                100 UDS
              </styled.span>
            </styled.div>
          </f1>,
          <f1
            validate={false}
          >
            <styled.div>
              No information on file. Complete new patient form below.
            </styled.div>
          </f1>,
        ],
        "frequency": "",
        "isUnsaved": false,
        "nextDeliveryDate": "",
        "prescriptionType": "permanent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": <styled.div>
              <styled.h1>
                <LocalizedText
                  textKey="prescription.title"
                />
              </styled.h1>
              <f1
                validate={true}
              >
                <styled.h2>
                  PRIMARY
                </styled.h2>
              </f1>
            </styled.div>,
            "isNested": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.h1>
                  <LocalizedText
                    textKey="prescription.title"
                  />
                </styled.h1>,
                <f1
                  validate={true}
                >
                  <styled.h2>
                    PRIMARY
                  </styled.h2>
                </f1>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <LocalizedText
                    textKey="prescription.title"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "textKey": "prescription.title",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": <styled.h2>
                    PRIMARY
                  </styled.h2>,
                  "validate": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "PRIMARY",
                  },
                  "ref": null,
                  "rendered": "PRIMARY",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <styled.div>
              <styled.span>
                Strip Consumption
              </styled.span>
              <styled.span>
                6 x Day
              </styled.span>
            </styled.div>,
            "validate": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  Strip Consumption
                </styled.span>,
                <styled.span>
                  6 x Day
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Strip Consumption",
                },
                "ref": null,
                "rendered": "Strip Consumption",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "6 x Day",
                },
                "ref": null,
                "rendered": "6 x Day",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <styled.div>
              <styled.span>
                Next delivery:
              </styled.span>
              <styled.span>
                
              </styled.span>
            </styled.div>,
            "validate": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  Next delivery:
                </styled.span>,
                <styled.span>
                  
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Next delivery:",
                },
                "ref": null,
                "rendered": "Next delivery:",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <styled.div>
              <styled.span>
                Model
              </styled.span>
              <styled.span>
                
              </styled.span>
            </styled.div>,
            "validate": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  Model
                </styled.span>,
                <styled.span>
                  
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Model",
                },
                "ref": null,
                "rendered": "Model",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <styled.div>
              <styled.span>
                Total Patient Stock
              </styled.span>
              <styled.span>
                100 UDS
              </styled.span>
            </styled.div>,
            "validate": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  Total Patient Stock
                </styled.span>,
                <styled.span>
                  100 UDS
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Total Patient Stock",
                },
                "ref": null,
                "rendered": "Total Patient Stock",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "100 UDS",
                },
                "ref": null,
                "rendered": "100 UDS",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <styled.div>
              No information on file. Complete new patient form below.
            </styled.div>,
            "validate": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "No information on file. Complete new patient form below.",
            },
            "ref": null,
            "rendered": "No information on file. Complete new patient form below.",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "children": <styled.div
          frequency=""
          isUnsaved={false}
          nextDeliveryDate=""
          prescriptionType="permanent"
        >
          <styled.div
            active={false}
            isNested={false}
          >
            <styled.div>
              <styled.h1>
                <LocalizedText
                  textKey="prescription.title"
                />
              </styled.h1>
              <f1
                validate={true}
              >
                <styled.h2>
                  PRIMARY
                </styled.h2>
              </f1>
            </styled.div>
          </styled.div>
          <f1
            validate={true}
          >
            <styled.div>
              <styled.span>
                Strip Consumption
              </styled.span>
              <styled.span>
                6 x Day
              </styled.span>
            </styled.div>
          </f1>
          <f1
            validate=""
          >
            <styled.div>
              <styled.span>
                Next delivery:
              </styled.span>
              <styled.span>
                
              </styled.span>
            </styled.div>
          </f1>
          <f1
            validate=""
          >
            <styled.div>
              <styled.span>
                Model
              </styled.span>
              <styled.span>
                
              </styled.span>
            </styled.div>
          </f1>
          <f1
            validate={true}
          >
            <styled.div>
              <styled.span>
                Total Patient Stock
              </styled.span>
              <styled.span>
                100 UDS
              </styled.span>
            </styled.div>
          </f1>
          <f1
            validate={false}
          >
            <styled.div>
              No information on file. Complete new patient form below.
            </styled.div>
          </f1>
        </styled.div>,
        "noPadding": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              active={false}
              isNested={false}
            >
              <styled.div>
                <styled.h1>
                  <LocalizedText
                    textKey="prescription.title"
                  />
                </styled.h1>
                <f1
                  validate={true}
                >
                  <styled.h2>
                    PRIMARY
                  </styled.h2>
                </f1>
              </styled.div>
            </styled.div>,
            <f1
              validate={true}
            >
              <styled.div>
                <styled.span>
                  Strip Consumption
                </styled.span>
                <styled.span>
                  6 x Day
                </styled.span>
              </styled.div>
            </f1>,
            <f1
              validate=""
            >
              <styled.div>
                <styled.span>
                  Next delivery:
                </styled.span>
                <styled.span>
                  
                </styled.span>
              </styled.div>
            </f1>,
            <f1
              validate=""
            >
              <styled.div>
                <styled.span>
                  Model
                </styled.span>
                <styled.span>
                  
                </styled.span>
              </styled.div>
            </f1>,
            <f1
              validate={true}
            >
              <styled.div>
                <styled.span>
                  Total Patient Stock
                </styled.span>
                <styled.span>
                  100 UDS
                </styled.span>
              </styled.div>
            </f1>,
            <f1
              validate={false}
            >
              <styled.div>
                No information on file. Complete new patient form below.
              </styled.div>
            </f1>,
          ],
          "frequency": "",
          "isUnsaved": false,
          "nextDeliveryDate": "",
          "prescriptionType": "permanent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": <styled.div>
                <styled.h1>
                  <LocalizedText
                    textKey="prescription.title"
                  />
                </styled.h1>
                <f1
                  validate={true}
                >
                  <styled.h2>
                    PRIMARY
                  </styled.h2>
                </f1>
              </styled.div>,
              "isNested": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.h1>
                    <LocalizedText
                      textKey="prescription.title"
                    />
                  </styled.h1>,
                  <f1
                    validate={true}
                  >
                    <styled.h2>
                      PRIMARY
                    </styled.h2>
                  </f1>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <LocalizedText
                      textKey="prescription.title"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "prescription.title",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": <styled.h2>
                      PRIMARY
                    </styled.h2>,
                    "validate": true,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "PRIMARY",
                    },
                    "ref": null,
                    "rendered": "PRIMARY",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <styled.div>
                <styled.span>
                  Strip Consumption
                </styled.span>
                <styled.span>
                  6 x Day
                </styled.span>
              </styled.div>,
              "validate": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    Strip Consumption
                  </styled.span>,
                  <styled.span>
                    6 x Day
                  </styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Strip Consumption",
                  },
                  "ref": null,
                  "rendered": "Strip Consumption",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "6 x Day",
                  },
                  "ref": null,
                  "rendered": "6 x Day",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <styled.div>
                <styled.span>
                  Next delivery:
                </styled.span>
                <styled.span>
                  
                </styled.span>
              </styled.div>,
              "validate": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    Next delivery:
                  </styled.span>,
                  <styled.span>
                    
                  </styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Next delivery:",
                  },
                  "ref": null,
                  "rendered": "Next delivery:",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <styled.div>
                <styled.span>
                  Model
                </styled.span>
                <styled.span>
                  
                </styled.span>
              </styled.div>,
              "validate": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    Model
                  </styled.span>,
                  <styled.span>
                    
                  </styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Model",
                  },
                  "ref": null,
                  "rendered": "Model",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <styled.div>
                <styled.span>
                  Total Patient Stock
                </styled.span>
                <styled.span>
                  100 UDS
                </styled.span>
              </styled.div>,
              "validate": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    Total Patient Stock
                  </styled.span>,
                  <styled.span>
                    100 UDS
                  </styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Total Patient Stock",
                  },
                  "ref": null,
                  "rendered": "Total Patient Stock",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "100 UDS",
                  },
                  "ref": null,
                  "rendered": "100 UDS",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <styled.div>
                No information on file. Complete new patient form below.
              </styled.div>,
              "validate": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "No information on file. Complete new patient form below.",
              },
              "ref": null,
              "rendered": "No information on file. Complete new patient form below.",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
